### DISCUSSION TITLE
Token count zero for streaming AzureOpenAI calls in langchain

### DISCUSSION THREAD

**Discussion by addirathod** [Arize-ai/phoenix#3315] [Q&A] [answered]
**Issue Description:** Whenever the streaming is set to True for the LLM initialization (see code below), the token count shows up as zero (as shown in the image).
<img width="1305" alt="Screenshot 2024-05-26 at 8 54 17 AM" src="https://github.com/Arize-ai/phoenix/assets/20735753/cd4042d5-a7a2-4c87-a1a7-ab79c939a267">

1. **Code 1: I have my LLM instance initialized in the following way:**

from langchain.callbacks import AsyncIteratorCallbackHandler
from langchain_openai import AzureChatOpenAI
def get_chain_details(streaming= True, deployment_name="gpt4-8k"):
    callback = AsyncIteratorCallbackHandler()
    LLM_BIG_STREAMING = AzureChatOpenAI(
        streaming=streaming,
        callbacks=[callback],
        temperature=0,
        azure_endpoint=AZURE_OPENAI_API_BASE,
        api_version="2023-05-15",
        azure_deployment=deployment_name,
        api_key=AZURE_OPENAI_API_KEY,
    )
    

2. **Code 2: And my instrumentation initialized like:**

import phoenix as px
session = px.launch_app()
from phoenix.trace.langchain import LangChainInstrumentor
LangChainInstrumentor().instrument()

3. **Versions**
Langchain: 0.1.12
Phoenix: 4.1.2

Will appreciate any help here Thanks

**Comment by RogerHYang** [comment, marked as answer]
Token counting for streaming is only [recently available](https://community.openai.com/t/usage-stats-now-available-when-using-streaming-with-the-chat-completions-api-or-completions-api/738156), so it's probably not in your API version, i.e. 2023-05-15.
